R/path.R

```diff
@@ -165,21 +165,31 @@ cmdstan_default_path <- function(old = FALSE, dir = NULL) {
 #' @param path Path to installation.
 #' @return Version number as a string.
 read_cmdstan_version <- function(path) {
-  makefile_path <- file.path(path, "makefile")
-  if (!file.exists(makefile_path)) {
-    warning(
-      "Can't find CmdStan makefile to detect version number. ",
-      "Path may not point to valid installation.",
-      call. = FALSE
-    )
-    return(NULL)
-  }
-  makefile <- readLines(makefile_path)
-  version_line <- grep("^CMDSTAN_VERSION :=", makefile, value = TRUE)
-  if (length(version_line) == 0) {
-    stop("CmdStan makefile is missing a version number.", call. = FALSE)
+  # Return all repo branch names if cmdstan installed from Github
+  if (dir.exists(file.path(path, ".git"))) {
+    cmdstan_ref <- get_current_git_ref(path)
+    stan_ref <- get_current_git_ref(file.path(path, "stan"))
+    math_ref <- get_current_git_ref(file.path(path, "stan", "lib", "stan_math"))
+
+
+    paste0("CmdStan: ", cmdstan_ref, ", Stan: ", stan_ref, ", Math: ", math_ref)
```

Comment:
We rely on the CmdStan version number in some parts of the code in cmdstanr (some features are only available after certain versions. I think this should return the version and then append the branch names.

So something like "2.30.0-develop/develop/some-branch" or something similar.